core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -102,6 +123,52 @@
         endOfStream = parent.isEndOfStream();
     }
 
+    private void copyCachedValues(HttpHeadersBase parant) {
+        method = parant.method;
+        status = parant.status;
+        contentType = parant.contentType;
+        contentDisposition = parant.contentDisposition;
+        acceptLanguages = parant.acceptLanguages;
+        cookies = parant.cookies;
+        setCookie = parant.setCookie;
+    }
+
+    @Override
+    void onChange(@Nullable AsciiString name) {
```

Comment:
Can't we implement this in `DefaultRequestHeaders` and `DefaultResponseHeaders` move down the request fields to `DefaultRequestHeaders` and the response fields to `DefaultResponseHeaders`?

Reply:
Your suggestion looks ideal. But I'd like to manage the cache values in one place so that we can see all cached easily.
If you are not strong here, I want to keep the current implementation for simplicity's sake. 
I will comment about common values and specific values for a request/response.